src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -88,6 +88,11 @@ public bool IsKeyword(SyntaxToken token)
                 SyntaxFacts.IsKeywordKind(kind); // both contextual and reserved keywords
         }
 
+        public bool IsReservedKeyword(string text)
+        {
+            return SyntaxFacts.GetKeywordKind(text) != SyntaxKind.None; // reserved keywords only
```

Comment:
this comment isn't helpful.  but we should document hte ISyntaxFactsService version.